module/mempool/stdmap/backdata/arrayBackData.go

```diff
@@ -0,0 +1,400 @@
+package backdata
+
+import (
+	"encoding/binary"
+	"time"
+	_ "unsafe" // for linking runtimeNano
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/mempool/stdmap/backdata/heropool"
+)
+
+//go:linkname runtimeNano runtime.nanotime
+func runtimeNano() int64
+
+const (
+	bucketSize = uint64(16)
+
+	// telemetryCounterInterval is the number of required interactions with
+	// this back data prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations. An interaction can be
+	// a read or write.
+	telemetryCounterInterval = uint64(1000)
+
+	// telemetryDurationInterval is the required elapsed duration interval
+	// prior to printing any log. This is done as a slow-down mechanism
+	// to avoid spamming logs upon read/write heavy operations.
+	telemetryDurationInterval = 10 * time.Second
+)
+
+// bIndex is data type representing a bucket index.
+type bIndex uint64
+
+// sIndex is data type representing a slot index in a bucket.
+type sIndex uint64
+
+// idPrefix is a 32-bits prefix flow.Identifier used to determine the bIndex of the entity
+// it represents.
+type idPrefix uint32
+
+type key struct {
+	keyIndex    uint64          // slot age.
+	entityIndex heropool.EIndex // link to actual entity.
+	idPref      idPrefix        // 32-bits prefix of entity identifier.
+}
+
+// keyBucket represents a bucket of keys.
+type keyBucket [bucketSize]key
+
+// ArrayBackData implements an array-based generic memory pool backed by a fixed total array.
+type ArrayBackData struct {
```

Comment:
Consider renaming `ArrayBackData` to `PoolByKey`, or just using `PoolByKey` internally if `ArrayBackData` needs to be used externally?

Reply:
`ArrayBackData` is the external representation of `herocache`, so I renamed it to `Cache` hence it will be accessible as `herocache.Cache` externally.

Reply:
Already replied to / addressed by another comment... :-)